src/components/Grid/GridCol.tsx

```diff
@@ -1,70 +1,36 @@
 import styled from 'styled-components';
 import { MOBILE_ONLY_MEDIA_QUERY, TABLET_MEDIA_QUERY, DESKTOP_MEDIA_QUERY } from '../../styles';
 
+type mobileValues = 1 | 2 | 3 | 4;
+type tabletDesktopValues = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
 interface Props {
-  desktopSpan?: number;
-  tabletSpan?: number;
-  mobileSpan?: number;
+  desktop?: boolean | tabletDesktopValues;
+  tablet?: boolean | tabletDesktopValues;
+  mobile?: boolean | mobileValues;
 }
 
-// const whiteColor: "#fff"
-// const CSS = `background-color: ${whiteColor};`
-/*
-    console.log(props)
-
-    props: {
-      desktopSpan: 3
-      tabletSpan:3
-      mobileSpan:3
-    }
-
-    {props} = destructuring
-    look into props, get the variables
-    instead of refering to prop variable, refer directly to variable inside prop
-
-    same thing as
-    ${props.mobileSpan}
-    */
-
 export const GridCol = styled.div<Props>`
   @media ${MOBILE_ONLY_MEDIA_QUERY} {
-    ${({ mobileSpan }) =>
-      `grid-column: auto / span ${mobileSpan};
-  `};
+    ${({ mobile }) =>
+      mobile
+        ? `grid-column: auto / span ${mobile};
+      -ms-grid-column: ${mobile};`
+        : 'display: none'};
   }
   @media ${TABLET_MEDIA_QUERY} {
-    ${({ tabletSpan }) =>
-      `grid-column: auto / span ${tabletSpan};
-      `};
+    ${({ tablet }) =>
+      tablet
+        ? `grid-column: auto / span ${tablet};
+      -ms-grid-column: ${tablet};`
+        : 'display:none'};
   }
   @media ${DESKTOP_MEDIA_QUERY} {
-    ${({ desktopSpan }) =>
-      `grid-column: auto / span ${desktopSpan};
-      `};
+    ${({ desktop }) =>
+      desktop
+        ? `grid-column: auto / span ${desktop};
+      -ms-grid-column: ${desktop};`
+        : 'display:none'};
+
+    ${({ desktop, tablet }) => desktop && !tablet && 'display: inline'}
   }
```

Comment:
what does this stuff do

Reply:
Basically this code is checking if the `GridCol` component received the `mobile`, `tablet`, or `desktop` prop.
If they received a `mobile, tablet, or desktop` prop, then render the amount of columns specific to that screen size.
So for your design, you want 3 columns on the left and 9 columns on the right for desktop.
But in tablet/mobile, you might want a completely different design, such as like 5 columns on the left, and 5 on the right.
This code allows you to define specific column sizes for different screen sizes to produce different designs.

That code that you highlighted is saying, if there is a desktop prop, but no tablet prop, then turn display on, since all of them are by default display: none unless they receive a prop value.

The reason why that is the case is because the Tablet Media Query is saying for all screen sizes from tablet to infinity, so that includes the desktop size in it... 

I just copied the media queries from my job - if we changed the tablet media query to only be from the tablet screen sizes 767px to 1023px, we wouldn't need that code at line 34.